src/github.com/mozilla-services/pushgo/simplepush/handlers_endpoint.go

```diff
@@ -337,16 +339,22 @@ func (h *EndpointHandler) deliver(cn http.CloseNotifier, uaid, chid string,
 			cancelSignal = cn.CloseNotify()
 		}
 		// Route the update.
+		startTime := timeNow().UTC()
 		delivered, _ = h.router.Route(cancelSignal, uaid, chid, version,
-			timeNow().UTC(), requestID, data)
+			startTime, requestID, data)
+		routingTime = timeNow().UTC().Sub(startTime)
 		if delivered {
+			// Routing succeeded.
+			h.metrics.Increment("router.broadcast.hit")
+			h.metrics.Timer("updates.routed.hits", routingTime)
 			return true
 		}
 	}
-	// If the device is not connected to this server, indicate whether routing
-	// was successful.
 	if !workerConnected {
-		return
+		// Device is not connected to this server and routing failed.
+		h.metrics.Increment("router.broadcast.miss")
+		h.metrics.Timer("updates.routed.misses", routingTime)
```

Comment:
Won't this lead to router metrics being emitted even if routing is off, or not used?